web/beacon-app/src/features/auth/components/Register/RegistrationForm.tsx

```diff
@@ -44,6 +44,10 @@ function RegistrationForm({ onSubmit }: RegistrationFormProps) {
   });
   const { touched, errors, values, getFieldProps, setFieldValue, isSubmitting } = formik;
 
+  useEffect(() => {
+    sessionStorage.setItem('newRegistrationEmail', values.email);
```

Comment:
Since we don't want to direct the user to the `/verify` page until after they click the verification link, this felt like the better option.